AdminWebsite/AdminWebsite/Helper/AdministratorRoleClaimsHelper.cs

```diff
@@ -0,0 +1,65 @@
+﻿using AdminWebsite.Security;
+using AdminWebsite.Services.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+
+namespace AdminWebsite.Helper
+{
+    public class AdministratorRoleClaimsHelper
+    {
+        private const string VhOfficerAdministratorClaimName = "IsVhOfficerAdministratorRole";
+        private const string CaseAdministratorClaimName = "IsCaseAdministratorRole";
+        private const string AdministratorClaimName = "IsAdministratorRole";
+        private const string UserCaseTypesClaimName = "UserCaseTypes";
+
+        private readonly IEnumerable<Claim> _administratorClaims;
+
+        public AdministratorRoleClaimsHelper(UserGroupData userGroupData)
+        {
+            IsVhOfficerAdministratorRole = userGroupData.UserRole == UserRole.VhOfficer.ToString();
+            IsCaseAdministratorRole = userGroupData.UserRole == UserRole.CaseAdmin.ToString();
+            IsAdministratorRole = IsVhOfficerAdministratorRole || IsCaseAdministratorRole;
+
+            _administratorClaims = new List<Claim>
+            {
+                new Claim(VhOfficerAdministratorClaimName, IsVhOfficerAdministratorRole.ToString()),
+                new Claim(CaseAdministratorClaimName, IsCaseAdministratorRole.ToString()),
+                new Claim(AdministratorClaimName, (IsVhOfficerAdministratorRole || IsCaseAdministratorRole).ToString()),
+                new Claim(UserCaseTypesClaimName, string.Join(",", userGroupData.CaseTypes ?? Enumerable.Empty<string>()))
+            };
+        }
+
+        public AdministratorRoleClaimsHelper(IEnumerable<Claim> claims)
+        {
+            var claimsList = claims.ToList();
+            var vhAdminRoleClaim = claimsList.FirstOrDefault(x => x.Type == VhOfficerAdministratorClaimName);
+            var caseAdminRoleClaim = claimsList.FirstOrDefault(x => x.Type == CaseAdministratorClaimName);
+            var adminRoleClaim = claimsList.FirstOrDefault(x => x.Type == AdministratorClaimName);
+            var userCaseTypes = claimsList.FirstOrDefault(x => x.Type == UserCaseTypesClaimName);
+
+            IsVhOfficerAdministratorRole = GetClaimBoolValue(vhAdminRoleClaim);
+            IsCaseAdministratorRole = GetClaimBoolValue(caseAdminRoleClaim);
+            IsAdministratorRole = GetClaimBoolValue(adminRoleClaim);
+            UserCaseTypes = userCaseTypes?.Value?.Split(",".ToCharArray(), StringSplitOptions.RemoveEmptyEntries) ??
```

Comment:
This is an interesting example from a coverage type of perspective :)

It looks like this is joining together quite a few test cases:
should_have_no_case_types_if_missing_case_types_claim
should_have_no_case_types_if_case_types_claim_is_null
should_have_a_list_of_case_types_if_claims_contains_comma_seperated_list_of_case_types

We could have a chat of these cases though, I'm thinking we could simplify it by making a few assumptions

Reply:
Good spot, I have created  unit test
1) should_have_no_case_types_if_missing_case_types_claim

should_have_a_list_of_case_types_if_claims_contains_comma_seperated_list_of_case_types is already covered.

should_have_no_case_types_if_case_types_claim_is_null -> value of claim can not be null.

Reply:
I have changed most of the tests to be more like testing the behavior, please check them out